File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        #split methodis used for removing whitespaces from starting 
        #and end oa a string
        s=s.strip()
        #string's word is converted into list by split method
        lis=list(s.split())
        # reverse the list by using slicing method
        lis=lis[::-1]
        # ater reversing join the list by using join method
        return " ".join(lis)

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        s=s.trim();
        String[]arr=s.split(" +");
        String ans="";
        int n=arr.length;
        for(int i=n-1;i>=0;i--){
            String st=arr[i].trim();
            if(i!=0){      
                ans+=st+" ";
            }else{
                ans+=st;
            }
        }
        
        return ans;
    }
}